### PR TITLE
User tests

### PR DESCRIPTION
**Pull request by KacFlor**

### CODE REVIEW

File: src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        hAdmin = new User("HAdmin",  encoder.encode("1234"), new Customer(), Role.ADMIN);
+    }
+
+    @DisplayName("JUnit test for getAllUsers method")
+    @Test
+    public void testGetAllUsers(){
+
+        User user3 = new User("Test3", "encodedPassword3", new Customer(), Role.USER);
+
+        given(userRepository.findAll()).willReturn(List.of(user1,user2,user3,hAdmin));
```
Comment (Ficzer) [inline src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java:63]: I am not familiar with this test framework, this one here is a mock right?

File: src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        List<User> result = userService.getAllUsers();
+
+        assertThat(result).isNotNull();
+        assertThat(result.size()).isEqualTo(4);
```
Comment (Ficzer) [inline src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java:68]: Maybe check also their names

File: src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        Authentication authentication = Mockito.mock(Authentication.class);
+        when(authentication.getPrincipal()).thenReturn(userDetails);
+        when(securityContext.getAuthentication()).thenReturn(authentication);
+
+        when(userRepository.findByLogin("Test1")).thenReturn(user1);
+
+        User currentUser = userService.getCurrentUser();
+
+        assertThat(currentUser).isNotNull();
+        assertThat(currentUser).isEqualTo(user1);
```
Comment (Ficzer) [inline src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java]: Won't it check only a reference value, you did not implement is equal method for user. Check also the values inside user pls, to be sure

File: src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    public void setup() {
+        PasswordEncoder encoder = new BCryptPasswordEncoder();
+        user1 = new User("Test1", "encodedPassword1", new Customer(), Role.USER);
+        user2 = new User("Test2", "encodedPassword2", new Customer(), Role.USER);
+        hAdmin = new User("HAdmin",  encoder.encode("1234"), new Customer(), Role.ADMIN);
+    }
+
+    @DisplayName("JUnit test for getAllUsers method")
+    @Test
+    public void testGetAllUsers(){
```
Comment (Ficzer) [inline src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java:59]: I prefere names like, CanGetAllUsers, or something like that, but there is not strong convention for it. Check https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea

File: src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java (hunk 2 above)
Comment (KacFlor) [reply src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java:68]: done

File: src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java (hunk 1 above)
Comment (KacFlor) [reply src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java:63]: yes this is mock

File: src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java (hunk 3 above)
Comment (KacFlor) [reply src/test/java/com/KacFlor/ShopSpring/service/UserServiceTest.java]: done

**Review by Ficzer — APPROVED** [review]

**Pull request merged**